pkg/providers/aws/cloudwatch.go

```diff
@@ -41,91 +42,47 @@ func NewCloudWatchClient(ctx context.Context, cfg *aws.Config) *cloudWatchClient
 }
 
 // Get the resource consumption of an ec2 instance
-func (e *cloudWatchClient) GetEC2Metrics(ca *cache.Cache, region string, interval time.Duration) ([]v1.Instance, error) {
-	instances := []v1.Instance{}
-	local := make(map[string]*v1.Instance)
-
+func (e *cloudWatchClient) GetEC2Metrics(
+	ctx context.Context,
+	cache *marshaler.Marshaler,
+	region string,
+	interval time.Duration,
+) error {
 	end := time.Now().UTC()
 	start := end.Add(-interval)
 
 	// Get the cpu consumption for all the instances in the region
-	cpuMetrics, err := e.getEC2CPU(region, start, end, interval)
+	err := e.getEC2CPU(ctx, cache, region, start, end, interval)
```

Comment:
We should maybe add the cache to the cloudwatch client so as not to pass it through everywhere, if we are using a pointer there shouldn't be any overhead